core/mempool/src/pool_task.rs

```diff
@@ -176,4 +183,73 @@ pub fn spawn_pool(
         future::ok(())
     });
     tokio::spawn(task);
+
+    // Receive transaction gossips
+    let pool5 = pool.clone();
+    let task = inc_tx_gossip_rx.for_each(move |tx_gossip| {
+        // TODO: verify signature
+        if let Err(e) = pool5.add_payload_with_author(tx_gossip.payload, tx_gossip.sender_id) {
+            warn!(target: "pool", "Failed to add payload from tx gossip: {}", e);
+        }
+
+        future::ok(())
+    });
+    tokio::spawn(task);
+
+    let pool6 = pool.clone();
+    let task = Interval::new_interval(gossip_tx_period)
```

Comment:
the code above:
```
    let pool2 = pool.clone();
    let task = Interval::new_interval(payload_announce_period)
        .for_each(move |_| {
            if let Some((authority_id, payload)) = pool2.prepare_payload_announce() {
``` 
is suppose to do exactly the same thing.
the idea ws to put logic of deciding who to announce/gossip what payloads in `prepare_payload_announce`